### ISSUE TITLE
USWDS - Input mask: Improve copy/paste functionality

### ISSUE THREAD

**Issue by mandylloyd** [uswds/uswds#6314]
## Summary

Some improvements to paste functionality are in progress in #6203 however it only covers cases where paste is triggered via keyboard. Paste events triggered through clicking remain an issue.

- Pasting behavior should be consistent across all event types (key up/down or click)
- Error message behavior should be consistent across all event types

The update should meet the standards outlined in the [Input mask requirements doc](https://docs.google.com/document/d/1kLHaCl5PceSByz0NvZqSc-ot7tSjcbe1EJPbBAZM7LY/edit#heading=h.fa3sv1ray42g) (🔒 Google docs).

## Problem Statement

The current input mask functionality exhibits inconsistent paste behavior and error messaging across different event types (keyboard and click). This leads to unexpected results when pasting text, including incorrect formatting and potential security vulnerabilities.

## Steps To Reproduce 

1. Open the input mask (PR has not been merged in yet) and paste text via keyboard
2. Observe the behavior and formatting of the pasted text.
3. Now paste using right click and select paste from the menu
4. Compare the behavior. It does not format as it should and the error message does not trigger either.

## Potential Solutions

1. Update the `handleCurrentValue` function to consider pasted text when calculating the value, as mentioned in the commented code below.
2. Fix the `strippedValue` function to return the correct value when pasting.
3. Refactor the `for-loop` in `handleCurrentValue` to prevent returning an empty string when pasting.

Other improvements would be:
 - implement a unified paste event handler that can handle both keyboard and click paste events, ensuring consistent behavior and formatting.

**Comment by mlloydbixal** [comment]
Leaving a note here to record how I was troubleshooting this.

- Updates need to be made to `const { value } = el;` to consider use of pasted text.
- `strippedValue()` needs to be fixed to not return an empty string when pasting.
- for loop needs to be refactored so `newValue` doesn't get returned as an empty string when pasting.

```
const handleCurrentValue = (el) => {
  const isCharsetPresent = el.dataset.charset;
  const placeholder = isCharsetPresent || el.dataset.placeholder;
  // const { value } = el;
  // force default to pastedText value
  const value = el.value || pastedText;
  console.log('value: ', value)
  const len = placeholder.length;
  const { matchType } = checkMaskType(placeholder, value);
  let newValue = "";
  let i;
  let charIndex;

  // force true so it doesn't return an empty string
  // const strippedVal = strippedValue(isCharsetPresent, value);
  const strippedVal = strippedValue(true, value);
  console.log('stripped value: ', strippedVal)
  for (i = 0, charIndex = 0; i < len; i += 1) {
    const isInt = isInteger(strippedVal[charIndex]);
    const isLet = isLetter(strippedVal[charIndex]);
    const matchesNumber = maskedNumber.indexOf(placeholder[i]) >= 0;
    const matchesLetter = maskedLetter.indexOf(placeholder[i]) >= 0;

    if (
      (matchesNumber && isInt) ||
      (isCharsetPresent && matchesLetter && isLet)
    ) {
      newValue += strippedVal[charIndex];
      charIndex += 1;
    } else if (
      (!isCharsetPresent && !isInt && matchesNumber) ||
      (isCharsetPresent &&
        ((matchesLetter && !isLet) || (matchesNumber && !isInt)))
    ) {
      // returns here on first pass with newValue = ""
      // so pasted value never gets formatted/filtered but exists in input field
      // and full placeholder also exists in input field
      return { newValue, matchType };
    } else {
      newValue += placeholder[i];
    }
    // Break if no characters are left and the pattern is non-special character
    if (strippedVal[charIndex] === undefined) {
      break;
    }
  }

  // If a pasted input is invalid, strip it
  if (newValue.length !== len && inputAddedByPaste) {
    newValue = stripPasteValue(newValue, pastedText);
  }

  return { newValue, matchType };
};

```

**Comment by mahoneycm** [comment]
For refinement:

- [ ] Add problem statement
- [ ] Add steps to reproduce
- [ ] Add potential solutions

**Comment by mejiaj** [comment]
@jdotyoon added it to next sprint for refinement.

**Comment by dct2023** [comment]
@mlloydbixal Let's touchbase on this to talk through next steps to review the refinement steps.

**Comment by amycole501** [comment]
This issue also relates to [the new WCAG 2.2 3.3.8 accessible authentication](https://www.w3.org/TR/WCAG22/#accessible-authentication-minimum)
